### ISSUE TITLE
getProducts not working with Android.

### ISSUE THREAD

**Issue by KuldeepAIP** [dooboolab/flutter_inapp_purchase#337]
### I have done initialisation and then try to get the products

Hi, getProducts not working in Android. below is my minimal code.

```
@override void initState() {
    asyncInitState();
    _getProduct();
    super.initState();
}

void asyncInitState() async {
    await FlutterInappPurchase.instance.initialize();
}

@override void dispose() async {
    super.dispose();
    await FlutterInappPurchase.instance.finalize();
}

void asyncInitState() async {
    await FlutterInappPurchase.instance.initialize();
}

Future _getProduct() async {
    Singleton.instance.showDefaultProgress();
    List<IAPItem> items =
        await FlutterInappPurchase.instance.getProducts(_productLists);
    print(items);
}
```

I am getting below exception at below line of code.

**Exception Line Code** ==>  `List<IAPItem> items =  await FlutterInappPurchase.instance.getProducts(_productLists);`

**Exception Description** ==> **PlatformException (PlatformException(getItemsByType, IAP not prepared. Check if Google Play service is available., , null))**

Can anyone please help me out with this.
Thanks

**Comment by dinesh-parmar** [comment]
Put _getProducts()

**Comment by aipxperts-developer** [comment]
Hi @dinesh-parmar  i also tried it. while putting _getProduct inside asyncInitState() no exception is there but also the IAPItem itemlist is blank.
![Screenshot 2021-12-02 at 7 13 48 PM](https://user-images.githubusercontent.com/51985803/144433860-e719395c-a3e8-46ab-a2af-04cae43232aa.png)

**Comment by dinesh-parmar** [comment]
Check that the _productLists contains valid product id as mentioned in your Google play console.

**Comment by dinesh-parmar** [comment]
Can you please send the code here?

**Comment by KuldeepAIP** [comment]
Hi @dinesh-parmar 
Here is the code.

```
final List<String> _productLists = [SubscriptionPackage.monthlyPackage];
  List<IAPItem> _arrProductsInIAP = [];

  @override
  void initState() {
    asyncInitState();
    _getProduct();
    super.initState();
  }

  @override
  void dispose() async {
    super.dispose();
    await FlutterInappPurchase.instance.finalize();
  }

  void asyncInitState() async {
    await FlutterInappPurchase.instance.initialize();
  }

  Future _getProduct() async {
    List<IAPItem> items =
        await FlutterInappPurchase.instance.getProducts(_productLists);
    print(items);
  }
```

**Comment by KuldeepAIP** [comment]
One more thing, when trying to fetch the product list using **in_app_purchase 2.0.0** dependency i got the products. Please find attached screenshot.

![Screenshot 2021-12-03 at 11 01 21 AM](https://user-images.githubusercontent.com/60171515/144550778-b63b927f-ba69-4dde-b2b8-c70dab333215.png)

**Comment by dinesh-parmar** [comment]
Put _getProducts() inside _asyncInitState. Try to run it asynchronously.


``final List<String> _productLists = [SubscriptionPackage.monthlyPackage];
  List<IAPItem> _arrProductsInIAP = [];

  @override
  void initState() {
    asyncInitState();
    super.initState();
  }

  @override
  void dispose() async {
    super.dispose();
    await FlutterInappPurchase.instance.finalize();
  }

  void asyncInitState() async {
    await FlutterInappPurchase.instance.initialize();
    await _getProduct();
  }

  Future _getProduct() async {
    List<IAPItem> items =
        await FlutterInappPurchase.instance.getProducts(_productLists);
    print(items);
  }
``

**Comment by dinesh-parmar** [comment]
Put _getProducts() inside _asyncInitState(). Try to run it asynchronously.

`final List _productLists = [SubscriptionPackage.monthlyPackage];
List _arrProductsInIAP = [];

@OverRide
void initState() {
asyncInitState();
super.initState();
}

@OverRide
void dispose() async {
super.dispose();
await FlutterInappPurchase.instance.finalize();
}

void asyncInitState() async {
await FlutterInappPurchase.instance.initialize();
await _getProduct();
}

Future _getProduct() async {
List items =
await FlutterInappPurchase.instance.getProducts(_productLists);
print(items);
}
`

**Comment by KuldeepAIP** [comment]
okay let me try this.

**Comment by dinesh-parmar** [comment]
Put _getProducts() inside _asyncInitState(). Try to run it asynchronously.

```
final List _productLists = [SubscriptionPackage.monthlyPackage];
List _arrProductsInIAP = [];

@OverRide
void initState() {
asyncInitState();
super.initState();
}

@OverRide
void dispose() async {
super.dispose();
await FlutterInappPurchase.instance.finalize();
}

void asyncInitState() async {
await FlutterInappPurchase.instance.initialize();
await _getProduct();
}

Future _getProduct() async {
List items =
await FlutterInappPurchase.instance.getProducts(_productLists);
print(items);
}

```

**Comment by KuldeepAIP** [comment]
If i put _getProducts() inside _asyncInitState(). then list is blank.
![Screenshot 2021-12-03 at 11 10 05 AM](https://user-images.githubusercontent.com/60171515/144551459-aca7f077-5a23-470b-a360-d7b72427e999.png)

**Comment by ArthurKnight** [comment]
in ios  getProduct() return 0 items.

**Comment by KuldeepAIP** [comment]
Hi @ArthurKnight  @dinesh-parmar 
Thanks for your support. I reviewed the documentation another time and i got to know this.
`getProducts()` returns the items in iOS but doesn't returns the items in Android. To get the items in Android need to use `getSubscriptions()` as per the doc.

**Comment by d9media** [comment]
Hey so after too many hours of bugfixing I experienced this issue with ... Docker (believe it or not!). I just noticed it becasue my CPU was maxed out and when I checked activity monitor, I could see Docker taking all resources. I have PiHole running. Apparently there is a collision between Flutter VM and Docker/PiHole. When I force quit Docker on MacOS, products loaded. This is so weird and random. I hope it helps
